src/iuse_actor.cpp

```diff
@@ -120,13 +125,12 @@ long iuse_transform::use(player *p, item *it, bool t, const tripoint &pos ) cons
     item *obj;
     if( container.empty() ) {
         obj = &it->convert( target );
+        if( ammo_qty >= 0 ) {
+            obj->ammo_set( ammo_type.empty() ? obj->ammo_current() : ammo_type, ammo_qty );
+        }
     } else {
         it->convert( container );
-        obj = &it->emplace_back( target );
-    }
-
-    if( ammo_qty >= 0 ) {
-        obj->ammo_set( ammo_type.empty() ? obj->ammo_current() : ammo_type, ammo_qty );
+        obj = &it->emplace_back( item( target, calendar::turn, ammo_qty >= 0 ? ammo_qty : 1 ) ) ;
```

Comment:
I don't think you need to explicitly call the constructor there
Possibly `std::max` rather than the ternary operator?